seacrowd/sea_datasets/lti_langid_corpus/lti_langid_corpus.py

```diff
@@ -0,0 +1,308 @@
+    "blz",
+    "mbi",
+    "ilo",
+    "mkn",
+    "isd",
+    "cth",
+    "bpr",
+    "por",
+    "mbb",
+    "tgl",
+    "msm",
+    "ivb",
+    "tam",
+    "plw",
+    "alp",
+    "row",
+]
+
+_LICENSE = Licenses.UNKNOWN.value
+
+_LOCAL = False
+
+_URL = "http://sourceforge.net/projects/la-strings/files/Language-Data/LTI-LangID-rel5.txz/download"
+
+# TODO: add supported task by dataset. One dataset may support multiple tasks
+_SUPPORTED_TASKS = [Tasks.LANGUAGE_IDENTIFICATION]
+
+_SOURCE_VERSION = "5.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+# TODO: Name the dataset class to match the script name using CamelCase instead of snake_case
+class LTILangIDDataset(datasets.GeneratorBasedBuilder):
+    """LTI LangID corpus is a dataset for language identification for 1266 languages."""
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_text",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features({"text": datasets.Value("string"), "language": datasets.Value("string")})
+
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text_features(label_names=_LANGUAGES)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(data_dir, "MIL-TALE", "5")
+
+        datasplits_dir = os.path.join(data_dir, "datasplits")
+        if not Path(datasplits_dir).exists() or not Path(os.path.join(datasplits_dir, "DONE")).exists():
+            # Run provided install.sh to generate train/dev/test splits in "datasplits" folder
+            subprocess.call([os.path.join(data_dir, "code", "install.sh"), datasplits_dir])
+            with open(os.path.join(datasplits_dir, "DONE"), "w"):
+                pass
+
+        train_filepaths = []
+        dev_filepaths = []
+        test_filepaths = []
+
+        for lang_id in _LANGUAGES:
+            train_filepaths.append(
+                (
+                    lang_id,
+                    glob(os.path.join(data_dir, "dataset", "train", f"{lang_id}*")),
```

Comment:
`glob(os.path.join(datasplits_dir, "train", f"{lang_id}*")),`

Reply:
@holylovenia I agree to add additional information to `_DESCRIPTION`